### PR TITLE
perf: EXC-1793: Add criterion load benchmark

### PR DESCRIPTION
**Pull request by berestovskyy**
The benchmark installs the specified number of `load_simulator` canisters using the `canister_creator` and executes the specified amount of rounds. It measures the total time taken to complete the rounds and the throughput, i.e., the number of rounds executed per second (essentially, the FR).

By default, each `load_simulator` canister runs a periodic timer with a one-second interval and accesses stable memory every fifth call.

This benchmark is useful for debugging and benchmarking scheduler and sandbox eviction changes. For more realistic testnet load tests, refer to the `dfinity/subnet-load-tester` project.

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

File: rs/rust_canisters/load_simulator/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Load Simulator Canister
+=======================
```
Comment (aterga) [inline rs/rust_canisters/load_simulator/README.md:2]: Is this canister only intended for testing? If so, I recommend mentioning that in the readme (not necessarily in this PR).

File: rs/rust_canisters/load_simulator/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+#[ic_cdk::query]
+fn no_op_query() {
+    // Do nothing.
+}
+
+#[ic_cdk::init]
+fn set_up_timer() {
+    // Set up a canister timer to call a function every N seconds.
+    ic_cdk_timers::set_timer_interval(Duration::from_secs(1), || ic_cdk::spawn(timer_handler()));
```
Comment (aterga) [inline rs/rust_canisters/load_simulator/src/main.rs:39]: Have you considered using a heartbeat vs. this timer? Maybe the difference doesn't matter for your use case, in which case please ignore this comment.

**Review by aterga — APPROVED** [review]

File: rs/rust_canisters/load_simulator/README.md (hunk 1 above)
Comment (berestovskyy) [reply rs/rust_canisters/load_simulator/README.md:2]: Added.

File: rs/rust_canisters/load_simulator/load_simulator.did [hunk 3]
```diff
@@ -0,0 +1 @@
+service : () -> { no_op_query : () -> () query; no_op_update : () -> () }
```
[the remark this answers is not in the document]
Comment (berestovskyy) [reply rs/rust_canisters/load_simulator/load_simulator.did]: This canister is a direct copy from the `subnet-load-tester` project. Both functions have the same effect and are currently unused in the benchmark. I'll remove them both for now...

File: rs/rust_canisters/load_simulator/src/main.rs (hunk 2 above)
Comment (berestovskyy) [reply rs/rust_canisters/load_simulator/src/main.rs:29]: It's simply to mimic a load on the subnet. Unlike heartbeats, timers also generate self-calls and replies, which better represents the load we see on some subnets.

**Pull request merged**